server/src/main/java/org/elasticsearch/rest/action/admin/indices/RestUpgradeAction.java

```diff
@@ -64,11 +64,12 @@ public String getName() {
     public RestChannelConsumer prepareRequest(RestRequest request, NodeClient client) throws IOException {
         if (request.method().equals(RestRequest.Method.GET)) {
             return handleGet(request, client);
-        } else if (request.method().equals(RestRequest.Method.POST)) {
+        } 
+        if (request.method().equals(RestRequest.Method.POST)) {
             return handlePost(request, client);
-        } else {
-            throw new IllegalArgumentException("illegal method [" + request.method() + "] for request [" + request.path() + "]");
         }
+        assert false;
+        return null;
```

Comment:
I would prefer not to use `assert false` because of the dummy return..

Reply:
That's fine; throw an `AssertionError` then. Also, I think that you can keep the existing structure: `if / else if / else`.

Reply:
Then the only change is actually the type of the exception which does not make it really simpler ...